ksql-engine/src/test/java/io/confluent/ksql/physical/PhysicalPlanBuilderTest.java

```diff
@@ -213,6 +214,39 @@ public void shouldMakeBareQuery() {
     assertThat(queryMetadata, instanceOf(QueuedQueryMetadata.class));
   }
 
+  @Test
+  public void shouldBuildTransientQueryWithCorrectSchema() {
+    // When:
+    final QueryMetadata queryMetadata = buildPhysicalPlan(simpleSelectFilter);
+
+    // Then:
+    assertThat(queryMetadata.getResultSchema(), is(
+        SchemaBuilder.struct()
+            .field("COL0", Schema.OPTIONAL_INT64_SCHEMA)
+            .field("COL2", Schema.OPTIONAL_STRING_SCHEMA)
+            .field("COL3", Schema.OPTIONAL_FLOAT64_SCHEMA)
+            .build()
+    ));
+  }
+
+  @Test
+  public void shouldBuildPersistentQueryWithCorrectSchema() {
+    // When:
+    final QueryMetadata queryMetadata = buildPhysicalPlan(
+        "CREATE STREAM FOO AS " + simpleSelectFilter);
+
+    // Then:
+    assertThat(queryMetadata.getResultSchema(), is(
+        SchemaBuilder.struct()
+            .field("ROWTIME", Schema.OPTIONAL_INT64_SCHEMA)
```

Comment:
Note: there is an inconsistency between transient and persistent queries as to whether ROWTIME and ROWKEY are in the schema or not.  This is existing functionality, (the test was added before changing the code).